code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -245,46 +272,321 @@
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/C = A
+
+	X.face_atom(C)
+
+	to_chat(X, "<span class='xenodanger'>We prepare to sting our quarry...</span>")
+
+	if(!do_after(X, HUNTER_STINGER_WINDUP, TRUE, target, BUSY_ICON_HOSTILE)) //Slight wind up
+		return fail_activate()
+
+	C.attack_alien(X) //We auto-sneak attack as part of this ability.
+	var/datum/reagent/toxin = /datum/reagent/toxin/acid/xeno_acid
+	var/transfer_amount = HUNTER_SNEAK_ATTACK_INJECT_AMOUNT
+	if(X.a_intent != INTENT_HARM) //Inject neurotoxin instead of acid while on disarm intent
+		toxin = /datum/reagent/toxin/xeno_neurotoxin
+
+	C.reagents.add_reagent(toxin, transfer_amount)
+	to_chat(C, "<span class='danger'>You feel a tiny prick.</span>") //Fluff
+	to_chat(X, "<span class='xenowarning'>Our stinger silently injects our victim!</span>")
+	X.playsound_local(C, 'sound/effects/spray3.ogg', 5, 0, 1)
+	C.playsound_local(C, 'sound/effects/spray3.ogg', 5, 0, 1)
+
+	succeed_activate()
+
+	GLOB.round_statistics.hunter_stings++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "hunter_stings") //Statistics
+	add_cooldown()
+
+
+// ***************************************
+// *********** Hunter's Mark
+// ***************************************
+/datum/action/xeno_action/activable/hunter_mark
+	name = "Hunter's Mark"
+	action_icon_state = "hunter_mark"
+	mechanics_text = "Psychically mark a creature you have line of sight to, allowing you to sense its direction, distance and location."
+	plasma_cost = 75
+	keybind_signal = COMSIG_XENOABILITY_HUNTER_MARK
+	cooldown_timer = 60 SECONDS
+
+/datum/action/xeno_action/activable/hunter_mark/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(QDELETED(A))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+	var/mob/living/C = A
+
+	if(!isliving(C))
+		to_chat(X, "<span class='xenowarning'>We cannot psychically mark this target!</span>")
+		return FALSE
+
+	if(C == X)
+		to_chat(X, "<span class='xenowarning'>Why would we target ourselves?</span>")
+		return FALSE
+
+	if(X.on_fire)
+		to_chat(X, "<span class='xenowarning'>We're too busy being on fire to mark them!</span>")
+		return FALSE
+
+	if(!X.line_of_sight(C)) //Need line of sight.
+		to_chat(X, "<span class='xenowarning'>We require line of sight to mark them!</span>")
+		return FALSE
+
+	return TRUE
+
+
+/datum/action/xeno_action/activable/hunter_mark/on_cooldown_finish()
+	var/mob/living/carbon/xenomorph/X = owner
+	to_chat(X, "<span class='xenowarning'><b>We are able to impose our psychic mark again.</b></span>")
+	X.playsound_local(X, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	return ..()
+
+
+/datum/action/xeno_action/activable/hunter_mark/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/M = A
+
+	if(!isliving(M)) //Sanity
+		to_chat(X, "<span class='xenodanger'>We cannot mark this target!</span>")
+		fail_activate()
```

Comment:
This check, is done 'twice' first in `can_use_ability()` if you're checking it here, have it runtime. `stack_trace()`

Otherwise, this check is unneeded and shouldn't be included.